modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/ParserServiceImplTest.java

```diff
@@ -49,54 +39,25 @@
  */
 public class ParserServiceImplTest {
     enum Statement {
-        QUERY("SELECT * FROM my_table", SqlQueryType.QUERY, true),
-        DML("INSERT INTO my_table VALUES (1, 1)", SqlQueryType.DML, true),
-        DDL("CREATE TABLE my_table (id INT PRIMARY KEY, avl INT)", SqlQueryType.DDL, false),
-        EXPLAIN_QUERY("EXPLAIN PLAN FOR SELECT * FROM my_table", SqlQueryType.EXPLAIN, false),
-        EXPLAIN_DML("EXPLAIN PLAN FOR INSERT INTO my_table VALUES (1, 1)", SqlQueryType.EXPLAIN, false);
+        QUERY("SELECT * FROM my_table", SqlQueryType.QUERY),
+        DML("INSERT INTO my_table VALUES (1, 1)", SqlQueryType.DML),
+        DDL("CREATE TABLE my_table (id INT PRIMARY KEY, avl INT)", SqlQueryType.DDL),
+        EXPLAIN_QUERY("EXPLAIN PLAN FOR SELECT * FROM my_table", SqlQueryType.EXPLAIN),
+        EXPLAIN_DML("EXPLAIN PLAN FOR INSERT INTO my_table VALUES (1, 1)", SqlQueryType.EXPLAIN);
 
         private final String text;
         private final SqlQueryType type;
-        private final boolean cacheable;
 
-        Statement(String text, SqlQueryType type, boolean cacheable) {
+        Statement(String text, SqlQueryType type) {
             this.text = text;
             this.type = type;
-            this.cacheable = cacheable;
-        }
-    }
-
-    @ParameterizedTest
-    @EnumSource(Statement.class)
-    void serviceAlwaysReturnsResultFromCacheIfPresent(Statement statement) {
-        ParsedResult expected = new DummyParsedResults();
-
-        ParsedResult actual = new ParserServiceImpl(0, new SameObjectCacheFactory(expected)).parse(statement.text);
-
-        assertSame(actual, expected);
-    }
-
-    @ParameterizedTest
-    @EnumSource(Statement.class)
-    void serviceCachesOnlyCertainStatements(Statement statement) {
```

Comment:
it looks like we are lost tests on caching

Reply:
that is true. Unfortunately, SqlQueryProcessor is a mess right. now, it's hard to cover it with proper unit tests

Reply:
could you please create a JIRA to cover it in the future

Reply:
https://issues.apache.org/jira/browse/IGNITE-21932